src/buffer/buffer.rs

```diff
@@ -124,23 +196,30 @@ impl Buffer {
     ///
     /// ```should_panic
     /// # use ratatui::prelude::*;
-    /// let rect = Rect::new(200, 100, 10, 10);
-    /// let buffer = Buffer::empty(rect);
+    /// let buffer = Buffer::empty(Rect::new(200, 100, 10, 10));
     /// // Top coordinate is outside of the buffer in global coordinate space, as the Buffer's area
     /// // starts at (200, 100).
     /// buffer.index_of(0, 0); // Panics
     /// ```
     #[track_caller]
     pub fn index_of(&self, x: u16, y: u16) -> usize {
```

Comment:
Given the low usage of `index_of` internally and externally I would consider changing it into `index_of_opt` directly.

https://github.com/search?q=ratatui+%22.index_of%28%22+AND+%28NOT+org%3Aratatui-org%29+AND+%28NOT+is%3Afork%29&type=code

Reply:
I'd prefer to make the `content` field and the methods that deal directly with indexes (`index_of`, `pos_of`) private as it may allow us to migrate to more efficient storage approaches than we currently use. For now, I'd prefer to keep this as an additive only change and do the larger things separately after some discussion.

Reply:
> I'd prefer to make the `content` field and the methods that deal directly with indexes (`index_of`, `pos_of`) private as it may allow us to migrate to more efficient storage approaches than we currently use.

The doc comment should tell the users then to not use this and to look at get_opt instead. Otherwise this will get lost in some PR comments and will not be read in the future by users of index_of.

Reply:
> The doc comment should tell the users then to not use this and to look at get_opt instead. Otherwise this will get lost in some PR comments and will not be read in the future by users of index_of.

Out of scope for this PR.

Reply:
Ended up being in scope